lib/api.ts

```diff
@@ -53,132 +22,158 @@ export interface MemberPublicEditing extends MemberPublic {
   editing?: { field: string; changeTo: string | string[] }[];
 }
 
-export async function getMembers(): Promise<MemberPublic[]> {
-  return Promise.all([
-    getBase({ name: "Members", view: "Approved" }),
-    getBase({ name: "Focuses", view: "Approved" }),
-    getBase({ name: "Industries", view: "Approved" }),
-    getBase({ name: "Regions" }),
-  ]).then((values) => {
-    const [members, focuses, industries, regions] = values;
-    return members
-      .map((member) => {
-        const regionLookup =
-          regions.find((region) => {
-            const memberRegion = member.fields["Region"];
-            if (memberRegion && Array.isArray(memberRegion)) {
-              return region.id === member.fields["Region"][0];
-            }
-          })?.fields["Name"] || null;
+export interface DocumentData {
+  id: string;
+  fields: any;
+}
 
-        const focusLookup = () => {
-          const memberFocus = member.fields["Focus"];
-          if (memberFocus && Array.isArray(memberFocus)) {
-            return memberFocus.map((foc) => {
-              return (
-                focuses
-                  .filter((thisFoc) => foc === thisFoc.id)
-                  .map((foc) => {
-                    return {
-                      name:
-                        typeof foc.fields["Name"] === "string"
-                          ? foc.fields["Name"]
-                          : null,
-                      id:
-                        typeof foc.fields["ID"] === "string"
-                          ? foc.fields["ID"]
-                          : null,
-                    };
-                  })[0] || null
-              );
-            });
-          }
-          return null;
-        };
+export async function getFirebaseTable(table: string) {
```

Comment:
Does it make sense to use an enum here? Since we'll have specific tables to fetch from, we may want to restrict what strings (e.g. "focuses", "members", etc)?

Reply:
hm, like an enum specifically for the expected table names? I like that idea, will implement upcoming